test/test.js

```diff
@@ -2530,6 +2533,7 @@ describe('Page', function() {
       expect(failedRequests[0].response()).toBe(null);
       expect(failedRequests[0].resourceType()).toBe('stylesheet');
       expect(failedRequests[0].failure().errorText).toBe('net::ERR_FAILED');
+      expect(failedRequests[0].frame()).toBeTruthy();
```

Comment:
Lets's also check somewhere that `request.frame()` returns reasonable frame for requests from subframes.
It would be nice to check this with and without request interception enabled.

Reply:
https://github.com/GoogleChrome/puppeteer/pull/1646/commits/77e7dd1e36b19eaeafe7668d2ed34dae6b662274

Reply:
Thank you. Can you please also add a test with a request from non-mainframe?

Reply:
Done!
https://github.com/GoogleChrome/puppeteer/pull/1646/commits/acf2c0942f541b243d293de6d196abe60512d965